models/RepPoints/point_ops.py

```diff
@@ -0,0 +1,322 @@
+import mxnet as mx
+import numpy as np
+
+
+def _gen_offsets(F, dcn_kernel=3, dcn_pad=1):
+    """ Generate offset for deformable convolutions.
+    :param dcn_kernel: the kernel of deformable convolutions
+    :param dcn_pad: the padding of deformable convolutions
+    :return: the offsets
+    """
+    dcn_base = F.arange(-dcn_pad, dcn_pad + 1)
+    dcn_base_y = F.repeat(dcn_base, dcn_kernel)
+    dcn_base_x = F.tile(dcn_base, dcn_kernel)
+    dcn_base_offset = F.reshape(F.stack(dcn_base_y, dcn_base_x, axis=1), (1, -1, 1, 1))
+    return dcn_base_offset
+
+
+def _gen_points(F, feat_map, stride):
```

Comment:
This may be moved to the weight processing stage as we did for the RPN part of FPN since it could be calculated once and cached for later usage. But the current implementation should be fine.

Reply:
I tried to cache the points but got no speedup. It seems that the points are not as dense as the anchors in RPN, and would not cause heavy overload on computation.